### PR TITLE
Update box refactor

### PR DESCRIPTION
**Pull request by clnoll**
Created txs for refactoring the update_contact_box txs.  Once these are reviewed the update_contact_box txs will also need refactoring.

### CODE REVIEW

File: app/assets/javascripts/angular/services/contact_service.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+app.factory("ContactService", ["$resource", "$rootScope", "ContactsBoxRsc",
```
Comment (sdevani) [inline app/assets/javascripts/angular/services/contact_service.js:1]: I'm ignoring this for this PR, it should not be part of the PR

File: app/services/Box/update_contact_box.rb [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# 1. Update contact's primary information
```
Comment (sdevani) [inline app/services/Box/update_contact_box.rb:1]: Start with an explanation of what's happening in english.

You're given a contact with new information, and this TXS updates it in the db.

File: app/services/Box/update_contact_info.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Updates a contact's information based off of
```
Comment (sdevani) [inline app/services/Box/update_contact_info.rb:1]: Looks good

File: app/services/Box/update_contact_stage.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      return failure(:invalid_pipeline_id)
+    end
+
+    if params[:stage_id].nil?
+      return failure(:no_stage_info_passed)
+    end
+
+    box = Box.where(contact_id: params[:contact_id], pipeline_id: params[:pipeline_id]).first
+
+    if box.nil?
```
Comment (sdevani) [inline app/services/Box/update_contact_stage.rb:32]: Error isn't properly described. There is no box id, so it cannot be invalid

File: app/services/Box/update_contact_stage.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      return failure(:invalid_box_id)
+    end
+
+    history = BoxHistory.create(box_id: box.id, stage_from: box.stage_id, stage_id: params[:stage_id])
```
Comment (sdevani) [inline app/services/Box/update_contact_stage.rb:36]: What if the stage_id did not change?

File: spec/services/Box/update_contact_field_values_spec.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    expect(result.error).to eq(:invalid_field_value_id)
+  end
+
+  it "fails when no boxfield hash is passed" do
+    result = UpdateContactFieldValues.run({contact_id: 1, field_id: field.id})
+    expect(result.success?).to eq(false)
+    expect(result.error).to eq(:no_field_value_info_passed)
+  end
+
+  it "updates the boxfield value given a field_values hash" do
```
Comment (sdevani) [inline spec/services/Box/update_contact_field_values_spec.rb:23]: Why does it need a field_id and a box_field id?

File: spec/services/Box/update_contact_field_values_spec.rb [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    result = UpdateContactFieldValues.run({contact_id: 1, field_id: field.id, field_values: {id: box_field.id, value: "updated"}})
+    expect(result.success?).to eq(true)
+    expect(result.boxfield.value).to eq("updated")
```
Comment (sdevani) [inline spec/services/Box/update_contact_field_values_spec.rb:26]: You should query the DB again to make sure it gives you an updated value

File: spec/services/Box/update_contact_note_spec.rb [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  it "fails when the note id is invalid" do
+    result = UpdateContactNote.run({note: {id: 1000}})
+    expect(result.success?).to eq(false)
+    expect(result.error).to eq(:invalid_note_id)
+  end
+
+  it "updates a note based on note id" do
+    result = UpdateContactNote.run({note: {id: note.id, notes: "update"}})
+    expect(result.success?).to eq(true)
+    expect(result.note.notes).to eq("update")
```
Comment (sdevani) [inline spec/services/Box/update_contact_note_spec.rb:22]: The result should have the note, but you should also make sure the update persisted in the DB

File: spec/services/Box/update_contact_stage_spec.rb [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+  it "fails when the pipeline_id is invalid" do
+    result = UpdateContactStage.run({contact_id: contact.id, pipeline_id: 1000})
+    expect(result.success?).to eq(false)
+    expect(result.error).to eq(:invalid_pipeline_id)
+  end
+
+  it "fails when the box id is invalid" do
+    result = UpdateContactStage.run({contact_id: contact2.id, pipeline_id: pipeline.id, stage_id: stage.id})
+    expect(result.success?).to eq(false)
+    expect(result.error).to eq(:invalid_box_id)
```
Comment (sdevani) [inline spec/services/Box/update_contact_stage_spec.rb:38]: Correct this error to mention there is no box between the pipeline and contact

File: spec/services/Box/update_contact_stage_spec.rb [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+
+  it "updates the contact box stage_id given a contact_id and pipeline_id" do
+    result = UpdateContactStage.run({contact_id: contact.id, pipeline_id: pipeline.id, stage_id: stage2.id})
+    expect(result.success?).to eq(true)
+    expect(result.box.stage_id).to eq(stage2.id)
+
+  end
+
+  it "updates the box_history with the former and current stage" do
+    result = UpdateContactStage.run({contact_id: contact.id, pipeline_id: pipeline.id, stage_id: stage2.id})
```
Comment (sdevani) [inline spec/services/Box/update_contact_stage_spec.rb:55]: Nice!

File: app/services/Box/update_contact_box.rb [hunk 11]
```diff
@@ -5,26 +9,18 @@ def run(params)
...
-      b.contact.city = params[:contact][:city] || b.contact.city
-    end
-
-    if params.include?(:status)
-      UpdateBoxStage.run(params)
-    end
-    # field = Field.where('pipeline_id = ?', params[:pipeline_id]).first
-    # field.field_name = params[:field_name] || field.field_name if !field.nil?
-    # field.field_type = params[:field_type] || field.field_type if !field.nil?
+    UpdateContactInfo.run(params)
```
Comment (sdevani) [inline app/services/Box/update_contact_box.rb:12]: We can revisit this one once you've cleaned it up and all

File: app/assets/javascripts/angular/services/contact_service.js (hunk 1 above)
Comment (clnoll) [inline app/assets/javascripts/angular/services/contact_service.js:1]: I concur.

File: app/services/Box/update_contact_box.rb (hunk 2 above)
Comment (clnoll) [inline app/services/Box/update_contact_box.rb:1]: Will resubmit once I address changes below.

File: app/services/Box/update_contact_info.rb (hunk 3 above)
Comment (clnoll) [inline app/services/Box/update_contact_info.rb:1]: Fin.

File: app/services/Box/update_contact_stage.rb (hunk 4 above)
Comment (clnoll) [inline app/services/Box/update_contact_stage.rb:32]: Edited.

File: app/services/Box/update_contact_stage.rb (hunk 5 above)
Comment (clnoll) [inline app/services/Box/update_contact_stage.rb:36]: Updated to not create BoxHistory record if the status hasn't changed

File: spec/services/Box/update_contact_field_values_spec.rb (hunk 6 above)
Comment (clnoll) [inline spec/services/Box/update_contact_field_values_spec.rb:23]: Edited.

File: spec/services/Box/update_contact_field_values_spec.rb (hunk 7 above)
Comment (clnoll) [inline spec/services/Box/update_contact_field_values_spec.rb:26]: Done.

File: spec/services/Box/update_contact_note_spec.rb (hunk 8 above)
Comment (clnoll) [inline spec/services/Box/update_contact_note_spec.rb:22]: Done.

File: spec/services/Box/update_contact_stage_spec.rb (hunk 9 above)
Comment (clnoll) [inline spec/services/Box/update_contact_stage_spec.rb:38]: Done.
